samples/nrf9160/modem_shell/src/link/link_api.c

```diff
@@ -890,3 +890,57 @@ void link_api_modem_info_get_for_shell(const struct shell *shell, bool connected
 #endif /* CONFIG_AT_CMD */
 	}
 }
+
+#define AT_CMD_RAI "AT%RAI?"
+#define AT_CMD_RAI_RESP_PARAM_COUNT 2
+#define AT_CMD_RAI_RESP_ENABLE_INDEX 1
+
+int link_api_rai_status(bool *rai_status)
+{
+	enum at_cmd_state state = AT_CMD_ERROR;
+	struct at_param_list param_list = { 0 };
+	char at_response_str[20];
+	int err;
+	int status;
+
+	if (rai_status == NULL) {
+		return -EINVAL;
+	}
+	*rai_status = false;
+
+	err = at_cmd_write(AT_CMD_RAI, at_response_str, sizeof(at_response_str), &state);
+	if (state != AT_CMD_OK) {
+		shell_error(shell_global, "Error state=%d, error=%d", state, err);
+		return -EFAULT;
+	}
+
+	err = at_params_list_init(&param_list, AT_CMD_RAI_RESP_PARAM_COUNT);
```

Comment:
The list allocated by `at_params_list_init()` is not freed using `at_params_list_free()`, so there's memory leak.

Reply:
Aargh, fixed.